File: number-of-arithmetic-triplets/number-of-arithmetic-triplets.py
class Solution:
    def arithmeticTriplets(self, nums: List[int], diff: int) -> int:
        cnt = 0 
        n = len(nums)
        for i in range(n):
            if nums[i]+ diff in nums and nums[i]+diff+diff in nums:
                cnt += 1

        return cnt 
        

File: 2442-number-of-arithmetic-triplets/number-of-arithmetic-triplets.cpp
class Solution {
public:
    int arithmeticTriplets(vector<int>& nums, int diff) {
        int n = nums.size() -1 ;
        int cnt = 0;
        for(int i = 0 ; i < n ; i++){
            int j = i+1 ; 
            int k = n ;
            while(j<k){

            
            int val = nums[j]-nums[i];
            int val2 = nums[k]-nums[j];
            if(val==diff && val2==diff){
                cnt++;
                j++;
                k--;
            }
            else if(val<diff){
                j++;
            }
            else{
                k--;
            }
        }
        }
        return cnt;
    }
};